### ISSUE TITLE
Sort the result of completion

### ISSUE THREAD

**Issue by elythyr** [phpactor/phpactor#1107]
Hi,

The result of a completion is not sorted.
When completing a class name it would be great to be able to sort the result by namespaces.
It's especially useful when completing a class name used in a lot of library, `Configuration` or `Collection` for example, in order to be able to retrieve easily in the list the one we want to import.

**Comment by elythyr** [comment]
Given the current implementation with the generators I wonder if it should not be up to the client to sort the results it was provided with.

Otherwise server side we have to load all candidates in memory in order to be able to sort them, which might result in poor performances when completing something with a lot of possible candidates (like classes).
We might be able to leverage the `LimitingCompletor` to ensure that we actually only sort part of the all candidates.
But then it might be awkward for the user because having a partial result like:
```
Namespace\A\Foo
Namespace\B\Foo
Namespace\D\Foo
```
Will let him believe there is no result in `Namespace\C`, but the same problem occurs if the client is responsible for the sorting...

Maybe we could had some kind of `SortedCompletor` that will be called before the `LimitingCompletor` and only sort if there is at least 3 characters provided to filter the result, but then sometimes the result will be sorted some times not.

Does anyone have a better idea ?

**Comment by dantleech** [comment]
Not sure if this is CoC, the indexer, or the new sorting algorithm (was it even applied to the LS?) but the sorting is quite impractical for the past months:

![image](https://user-images.githubusercontent.com/530801/103406534-e49fa000-4b52-11eb-9b4a-bb989c49149b.png)

The most relevant class (i.e. the one defined in the current project) is last.
This is even worse for things like `RuntimeException` where the project class is at position 20 or so.

Having a sorting algorithm which can take into account the relevance of namespaces, some options:

1. Classes defined in-project get higher precedence (as determined by file path or namespace)
2. Sort classes by relatedness to the current namespace (not sure how that would work)

**Comment by BladeMF** [comment]
The result in VS Code is the same, although the one I am interested in is usually first.

**Comment by camilledejoye** [comment]
@dantleech your comment was before you PR about sorting suggestions ?

I'm not a big fan of the new CoC list neither, I often have the same class from different libraries and we can't see the namespaces directly anymore...
But it's nice when the completion popup is on the side of the screen because it's shown on the opposite side, which avoid having part of the popup outside of the screen. I guess that's why they did it but I still need to find a way to disable it :D

**Comment by dantleech** [comment]
Yes, the comment was before that merge, though the algorithm used isn't that effective, so could still be improved. Re. CoC I'd really like to find the config option to revert that change :smile:

**Comment by camilledejoye** [comment]
@dantleech adding `"suggest.floatEnable": false` to the CoC configuration seems to have done the trick.
I don't know what the impacts could be so we will have to try :)
